### ISSUE TITLE
Overly complicated @Config usage in Dynamic Darkness (Fix included)

### ISSUE THREAD

**Issue by Rongmario** [yamporg/mods#22]
[Here's a way to use @Config even if your mod is a coremod/mixin.](https://github.com/LoliKingdom/LoliASM/blob/b348eeb534d4bba954d792a876350f49120dfca5/src/main/java/zone/rong/loliasm/config/LoliConfig.java#L73)

**Comment by democat3457** [comment]
Unrelated but
> loli 

👀

**Comment by tie** [comment]
Yeah, it’s a rather dirty hack indeed. Thanks for giving a good reference that uses public API!

I wanted to keep the config GUI when I’ve dropped dependency on mixin. The current implementation is, uh, a temporary workaround since both config I/O and GUI require a decent amount of boilerplate, and `@Config` transparently handles both.

**Comment by Rongmario** [comment]
Yeah that's fair. I only opened this thread bc someone was getting a crash using this mod and my [LoliASM](https://github.com/LoliKingdom/LoliASM), which optimizes some strings in the background.



> Unrelated but
> 
> > loli
> 
> 👀

Yes.
